src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -740,85 +741,128 @@ private IntermediateNode InjectCoercion(IntermediateNode child, IRTranslatorCont
                 return new UnaryOpNode(IRContext.NotInSource(FormulaType.Build(toType)), unaryOpKind, child);
             }
 
-            private static IntermediateNode GetAddBinaryOp(IRTranslatorContext context, TexlBinaryOpNode node, IntermediateNode left, IntermediateNode right, DType leftType, DType rightType)
+            private static IntermediateNode GetAddSubBinaryOp(IRTranslatorContext context, TexlBinaryOpNode node, IntermediateNode left, IntermediateNode right, DType leftType, DType rightType)
             {
                 Contracts.AssertValue(node);
-                Contracts.Assert(node.Op == BinaryOp.Add);
+                Contracts.Assert(node.Op == BinaryOp.Add || node.Op == BinaryOp.Sub);
 
                 switch (leftType.Kind)
                 {
                     case DKind.Date:
                         if (rightType == DType.DateTime || rightType == DType.Date)
                         {
-                            // Date + '-DateTime' => in days
-                            // Date + '-Date' => in days
+                            Contracts.Assert(node.Op == BinaryOp.Sub);
```

Comment:
Consider changing these asserts to Check. Assert only runs in debug build, check runs in all builds.

Reply:
Done.